spec/integration/product-manager.spec.js

```diff
@@ -0,0 +1,114 @@
+import _ from 'lodash'
+import { expect } from 'chai'
+
+import * as utils from '../utils/helper'
+import ProductManager from '../../lib/services/product-manager'
+
+const mockProduct = {
+  name: {
+    en: 'Test Product'
+  },
+  slug: {
+    en: 'testProduct'
+  },
+  masterVariant: {
+    sku: '123'
+  },
+  productType: {
+    id: null,
+    typeId: 'product-type'
+  }
+}
+
+let ctpClient = null
+let productService = null
+let productType = null
+
+function getProductMock () {
+  return _.cloneDeep(mockProduct)
+}
+
+describe('ProductManager', () => {
+  before(async () => {
+    ctpClient = await utils.createClient()
+    await utils.deleteResource(ctpClient.products)
+    productType = await utils.ensureProductType(ctpClient)
+    mockProduct.productType.id = productType.id
+  })
+
+  beforeEach(() =>
+    productService = new ProductManager(utils.logger, ctpClient)
+  )
+
+  afterEach(() =>
+    utils.deleteResource(ctpClient.products)
+  )
+
+  describe('Basic functions', () => {
```

Comment:
I would not define all those tests in this file as integration but rather unit tests so save API calls and time. Those methods will never be used on their own but as whole in particular business use cases and those business use cases should be tested as integration, wdyt?

Reply:
I think that the business use cases should be definitely covered by integration tests and a vast majority of service methods should be covered by unit tests (at least where it makes sense).

But in more complicated methods (eg. [removing variants from a product](https://github.com/commercetools/commercetools-node-variant-reassignment/pull/15/files#diff-15ce2e3c2d77d574a8c8ae56d19b9d92R105)) can be plenty of edge cases which can break in the future (for example an order of API commands matters - removing current variant after staged works but not in the reversed order and etc.).

With one or two business tests you won't cover all these edge cases. Also, it does not make sense to write multiple business integration tests and repeatedly test other parts of the system (eg. loading products by SKU) to test the different behavior of `removeVariantsFromProduct`method.

Another reason for integration tests is that sometimes we can reuse services somewhere else so it is nice to have one fully tested and independent component which can be just copied with all tests.

Therefore I would make also some integration tests for services.

But I agree that there are some pointless tests which can be removed so I created this issue which should be taken as one of the last - https://github.com/commercetools/commercetools-node-variant-reassignment/issues/16